NebulaHost/PacketProcessors/Logistics/StationUIInitialSyncRequestProcessor.cs

```diff
@@ -0,0 +1,72 @@
+﻿using NebulaModel.Attributes;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Logistics;
+using NebulaModel.Packets.Processors;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace NebulaHost.PacketProcessors.Logistics
+{
+    [RegisterPacketProcessor]
+    public class StationUIInitialSyncRequestProcessor: IPacketProcessor<StationUIInitialSyncRequest>
+    {
+        public void ProcessPacket(StationUIInitialSyncRequest packet, NebulaConnection conn)
+        {
+            StationComponent[] gStationPool = GameMain.data.galacticTransport.stationPool;
+            if(packet.stationGId < gStationPool.Length)
+            {
+                StationComponent stationComponent = gStationPool[packet.stationGId];
+                if(stationComponent != null)
+                {
+                    List<int> itemId = new List<int>();
```

Comment:
Is this sending data for entries that aren't set?

Reply:
Also, you can pre-allocate arrays here.

```csharp
var storage = storageComponent.storage;
var itemId = new int[storage.Length];
...

for (int i = 0; i < storage.Length; ++i) {
    var entry = storage[i];
    itemId[i] = entry.itemId;
    itemCountMax = entry.max;
    ..
}

Reply:
Ah thanks, i should put the sending code inside the check.

Reply:
preallocation is done in next commit :)